pennylane/interfaces/tf.py

```diff
@@ -98,10 +101,24 @@ def __repr__(self):
         metric_tensor = qnode.metric_tensor
         draw = qnode.draw
         func = qnode.func
+        num_variables = property(lambda self: qnode.num_variables)
+        arg_vars = property(lambda self: qnode.arg_vars)
 
     @qnode_str
     @tf.custom_gradient
     def _TFQNode(*input_, **input_kwargs):
+        # Determine which input tensors/Variables are being recorded for backpropagation.
+        # The function should_record_backprop, documented here:
+        # https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python/eager/tape.py#L163
+        # accepts lists of *tensors* (not Variables), returning True if all are being watched by one or more
+        # existing gradient tape, False if not.
+        requires_grad = [
+            tape_lib.should_record_backprop([tf.convert_to_tensor(i)])
+            if isinstance(i, (Variable, tf.Tensor))
+            else False
+            for i in input_
+        ]
```

Comment:
I think I'm getting confused here, why do we care only about the input arguments and not keyword arguments?

Reply:
> The QNode parameter logic/signature introspection remains, so only positional arguments remain differentiable in the TF interface. Once the tensor refactor is complete, we can return to the QNode and see what can be deleted. If we are able to make keyword arguments differentiable in JacobianQNode.jacobian, we can extend all interfaces likewise.

I should read the PR description :laughing: